log/json_logger_test.go

```diff
@@ -0,0 +1,48 @@
+package log_test
+
+import (
+	"bytes"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/peterbourgon/gokit/log"
+)
+
+func TestJSONLogger(t *testing.T) {
+	buf := &bytes.Buffer{}
+	logger := log.NewJSONLogger(buf)
+
+	logger.Log("a")
+	if want, have := `{"msg":"a"}`+"\n", buf.String(); want != have {
+		t.Errorf("want %#v, have %#v", want, have)
+	}
+
+	buf.Reset()
+	if err := logger.With("level", "INFO").Log("b"); err != nil {
```

Comment:
I'd try to log `With()` various types to test that `error` special case, and other possible special cases =P

It's very frustrating to `log.With("err", err)` and see an empty object in the logs. =]